at-dungnguyen/Final/main.rb

```diff
@@ -0,0 +1,399 @@
+    when 2
+      @flag_team = false
+    when 3
+      @flag_team = false
+      @user = nil
+    when 4
+      @tc.add_member(@user['team']) if @user['role'] == SessionController::LEADER
+    when 5
+      @tc.add_team if @user['role'] == SessionController::ADMIN
+    when 0
+      @flag_team = false
+      execute2(0)
+    end
+  end
+
+  # Norification
+  def notification_management
+    p 'Notification Management'
+    p '======================='
+    p '1. List Notification'
+    p '2. Read a notification'
+    p '3. Back'
+    p '4. Logout'
+    p '5. Send a notification' if @user['role'] == SessionController::LEADER
+    p '6. Delete a notification' if @user['role'] == SessionController::LEADER
+    p '0. Exit'
+  end
+
+  def execute_notify(function)
+    case function
+    when 1
+      all_notify
+    when 2
+      g_notify
+    when 3
+      @flag_notify = false
+    when 4
+      @flag_notify = false
+      @user = nil
+    when 5
+      send_notify
+    when 6
+      del_notify
+    when 0
+      @flag_notify = false
+      execute2(0)
+    end
+  end
+
+  def all_notify
+    if @user['team']
+      list_notify = @nc.get_items(@user['team'])
+      if list_notify.any?
+        list_notify.each do |notify|
+          puts "#{notify['id']} : #{notify['title']} | #{notify['publish_date']}"
+        end
+      end
+    end
+    p '====================================================================='
+  end
+
+  def g_notify
+    p 'Input id notification'
+    id = input
+    notify = @nc.get_item(id, @user)
+    clear
+    if notify
+      p "#{notify['id']} : #{notify['title']} | #{notify['publish_date']}"
+      p notify['content']
+    else
+      p 'Invalid ID'
+    end
+    p '================================='
+  end
+
+  def send_notify
+    p 'Input title'
+    title = gets.chomp
+    p 'Input content'
+    content = gets.chomp
+    publish_date = Time.now
+    id_team = @user['team']
+    notify = Notification.new(title, content, publish_date.strftime('%F %T'), id_team)
+    clear
+    if @nc.add_item(notify)
+      p 'Send notify success'
+    else
+      p 'Send error!'
+    end
+    p '================================'
+  end
+
+  def del_notify
+    p 'Input id notification'
+    id = input
+    clear
+    if @nc.del_item(id)
+      p 'Delete Notification Success'
+    else
+      p "Notification doesn't exist!"
+    end
+    p '================================'
+  end
+
+  # Leave
+  def leave_management
+    p 'Leave Management'
+    p '==============='
+    p '1. Send a leave request'
+    p '2. Update a leave request'
+    p '3. Delete a leave request'
+    p '4. Show list of leaves of employees.'
+    p '5. Back'
+    p '6. Logout'
+    p '7. Show list leaves request need approve' if @user['role'] == SessionController::LEADER
+    p '8. Approve or reject leave request' if @user['role'] == SessionController::LEADER
+    p '0. Exit'
+  end
+
+  def execute_leave(function)
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/atruby-team/internship-18-summer/blob/3c0bf6c6d3420a006a4c5cda03d7111306060898/at-dungnguyen/Final/main.rb#L220"></span>Metrics/CyclomaticComplexity: Cyclomatic complexity for execute_leave is too high. [12/10]</td>
    </tr>
  </tbody>
</table>